File: Data Structures & Algorithms/jump-game-ii/submission-2.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        cache = [float('inf') for i in range(len(nums))]
        cache[-1] = 0
        cnt = -1
        i = 0
        l, r = 0, 0
        while l < len(nums):
            print(l, r)
            new_max = l
            for j in range(l, r+1):

                new_max = max(new_max, j+nums[j] if j < len(nums) else len(nums)-1)

            if new_max == l and l!= len(nums)-1:
                return -1
            l = r+1
            r = new_max
            cnt += 1
        return cnt


        

File: Data Structures & Algorithms/climbing-stairs/submission-2.cpp
class Solution {
public:
    int climbStairs(int n) {
        // dp[i] = dp[i-1] + dp[i-2]
        int a = 1, b = 2;
        for(int i = 3; i <= n; i++){
            int c = a + b;
            a = b;
            b = c;
        }
        if(n <= 1) return 1;
        return b;
    }
};
